src/store/assets/assets-epics.ts

```diff
@@ -1,19 +1,34 @@
+import { combineEpics } from 'redux-observable';
 import { Observable, of } from 'rxjs';
 import { catchError, map, switchMap } from 'rxjs/operators';
 import { Action } from 'ts-action';
 import { ofType, toPayload } from 'ts-action-operators';
 
-import { getAssetsRequest$ } from '../../api.service';
-import { loadAssetsActions } from './assets-actions';
+import { getAssetsRequest$, getTezosBalanceRequest$ } from '../../api.service';
+import { loadTezosAssetsActions, loadTokenAssetsActions } from './assets-actions';
 
-export const loadAssetsEpic = (action$: Observable<Action>) =>
+const loadTokenAssetsEpic = (action$: Observable<Action>) =>
   action$.pipe(
-    ofType(loadAssetsActions.submit),
+    ofType(loadTokenAssetsActions.submit),
     toPayload(),
     switchMap(payload =>
       getAssetsRequest$(payload).pipe(
-        map(response => loadAssetsActions.success(response.data.balances)),
-        catchError(err => of(loadAssetsActions.fail(err.response.data.message)))
+        map(response => loadTokenAssetsActions.success(response.data.balances)),
+        catchError(err => of(loadTokenAssetsActions.fail(err.response.data.message)))
       )
     )
   );
+
+const loadTezosAssetsEpic = (action$: Observable<Action>) =>
+  action$.pipe(
+    ofType(loadTezosAssetsActions.submit),
+    toPayload(),
+    switchMap(payload =>
+      getTezosBalanceRequest$(payload).pipe(
+        map(response => loadTezosAssetsActions.success(response.toString())),
+        catchError(err => of(loadTezosAssetsActions.fail(err.response.data.message)))
+      )
+    )
+  );
```

Comment:
Rewrite `getTezosBalanceRequest$` trying to avoid mixing of reactive and functional programing.
```
export const getTezosBalanceRequest$ = () => (address$: Observable<string>) =>
  address.pipe(
      withLatestFrom(tezos$),
      switchMap(([address, tezos]) => tezos.tz.getBalance(address))
  );
  
const loadTezosAssetsEpic = (action$: Observable<Action>) =>
  action$.pipe(
    ofType(loadTezosAssetsActions.submit),
    toPayload(),
    getTezosBalanceRequest$(),
    map(response => loadTezosAssetsActions.success(response.toString())),
    catchError(err => of(loadTezosAssetsActions.fail(err.response.data.message)))
    )
  );
```